src/Shared/Process/ProcessEx.cs

```diff
@@ -179,7 +179,7 @@ private void OnProcessExited(object sender, EventArgs e)
                 _output.WriteLine("Process exited.");
             }
         }
-        _process.WaitForExit();
+
```

Comment:
We're already in the `Exited` Event so this call is redundant:

> There are two ways of being notified when the associated process exits: synchronously and asynchronously. Synchronous notification means calling the WaitForExit method to block the current thread until the process exits. Asynchronous notification uses the Exited event, which allows the calling thread to continue execution in the meantime. In the latter case, EnableRaisingEvents must be set to true for the calling application to receive the Exited event.

https://learn.microsoft.com/dotnet/api/system.diagnostics.process.exited?view=net-7.0

Reply:
Pretty sure this is still needed when using async IO. It guarantees that stdout and stderr are done.
Data point; after this PR was merged there are now failures like:
```
2023-02-14T17:40:24.0586118Z [xUnit.net 00:00:04.23]         | [3.247s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: Process exited.
2023-02-14T17:40:24.0586151Z [xUnit.net 00:00:04.23]         | [3.248s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] No templates or subcommands found matching: 'reactredux'.
2023-02-14T17:40:24.0586182Z [xUnit.net 00:00:04.23]         | [3.248s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.0586210Z [xUnit.net 00:00:04.23]         | [3.248s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] To list installed templates similar to 'reactredux', run:
2023-02-14T17:40:24.0594633Z [xUnit.net 00:00:04.23]         | [3.248s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR]    dotnet new list reactredux
2023-02-14T17:40:24.0627195Z [xUnit.net 00:00:04.23]         | [3.248s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] To search for the templates on NuGet.org, run:
2023-02-14T17:40:24.0629366Z [xUnit.net 00:00:04.23]         | [3.248s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR]    dotnet new search reactredux
2023-02-14T17:40:24.0736529Z [xUnit.net 00:00:04.23]         | [3.248s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.0754965Z [xUnit.net 00:00:04.23]         | [3.248s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] 
2023-02-14T17:40:24.0774833Z [xUnit.net 00:00:04.23]         | [3.248s] Templates.Blazor.Test.BlazorWasmTemplateTest Information: [ERROR] For details on the exit code, refer to https://aka.ms/templating-exit-codes#103
2023-02-14T17:40:24.0794693Z [xUnit.net 00:00:04.23]         | [3.254s] Templates.Blazor.Test.BlazorWasmTemplateTest Error: Test threw an exception.
2023-02-14T17:40:24.0812567Z [xUnit.net 00:00:04.23]         | System.InvalidOperationException: Failed to uninstall previous templates. The template 'reactredux' could still be found.
2023-02-14T17:40:24.0901861Z [xUnit.net 00:00:04.23]         |    at Templates.Test.Helpers.TemplatePackageInstaller.VerifyCannotFindTemplateAsync(ITestOutputHelper output, String templateName) in /_/src/ProjectTemplates/Shared/TemplatePackageInstaller.cs:line 151
2023-02-14T17:40:24.0920908Z [xUnit.net 00:00:04.23]         |    at Templates.Test.Helpers.TemplatePackageInstaller.InstallTemplatePackages(ITestOutputHelper output) in /_/src/ProjectTemplates/Shared/TemplatePackageInstaller.cs:line 114
```
You can see from the test output that the text "No templates or subcommands found matching:" was output, however the test threw on the check for that exact string:
https://github.com/dotnet/aspnetcore/blob/fb1e6a74373f1e4f09cc1b0adaac5f0535e5f07d/src/ProjectTemplates/Shared/TemplatePackageInstaller.cs#L149-L151
This implies a race condition where the process exits and we grab the output before the stdout/stderr completed writing.
cc @wtgodbe

Reply:
Thanks for catching this, I'll put up a PR to revert this line